extensions/parquet/table/src/main/java/io/deephaven/parquet/table/util/TrackedSeekableChannelsProvider.java

```diff
@@ -34,9 +35,20 @@ public static SeekableChannelsProvider getInstance() {
         return instance;
     }
 
+    public static SeekableChannelsProvider getInstance(@NotNull final TrackedFileHandleFactory fileHandleFactory) {
+        if (instance == null) {
+            synchronized (TrackedSeekableChannelsProvider.class) {
+                if (instance == null) {
+                    return instance = new TrackedSeekableChannelsProvider(fileHandleFactory);
+                }
+            }
+        }
+        return instance;
+    }
+
     private final TrackedFileHandleFactory fileHandleFactory;
 
-    public TrackedSeekableChannelsProvider(@NotNull final TrackedFileHandleFactory fileHandleFactory) {
+    private TrackedSeekableChannelsProvider(@NotNull final TrackedFileHandleFactory fileHandleFactory) {
```

Comment:
I wonder if we should be integration the invalidation into this class, instead of the factory,

Reply:
The `CachedChannelProvider` instances also need to be invalidated so that they cannot allocate new channels. I think we want to start using a map from File (canonicalized) to WeakReference<CachedChannelProvider>. The provider could then start keeping track of all outstanding handles, rather than just the pooled ones.